### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'modified'

### DISCUSSION THREAD

**Discussion by Relebohile3369** [XeroAPI/xero-python#119] [Q&A]
Hi, 

I need help, I'm encountering this error below when I run my app on localhost:5000. Can you please help me to fix this error? 

"
File "/Users/lesmacbookpro/Desktop/Xero/THE  RIGHT ONE/xero-python-oauth2-starter-master/venv/lib/python3.11/site-packages/flask_session/sessions.py", line 343, in save_session
    if session.modified:
       ^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'modified'
127.0.0.1 - - [19/Aug/2023 18:12:19] "GET / HTTP/1.1" 500 -"

**Comment by SherryFeng123** [comment]
What worked for me was installing an older version of flask. The version I did was flask==2.2.5

**Comment by CedricConday** [comment]
This isn't actually a xero-python issue — the traceback is in `flask_session/sessions.py`, so it's a **Flask / Flask-Session version mismatch**. `if session.modified` fails because `open_session()` returned `None`, which happens when your installed **Flask-Session** doesn't support your installed **Flask** (Flask 2.3 removed internals that older Flask-Session releases relied on).

Two clean fixes:

- **Upgrade Flask-Session** to a modern release: `pip install -U Flask-Session` (0.5.0+ restored compatibility with current Flask) — preferred, keeps you up to date.
- Or **pin Flask** below the break, which is what @SherryFeng123's `flask==2.2.5` effectively does.

Pinning Flask down works but leaves you on an old Flask; upgrading Flask-Session is the better long-term fix. After switching, clear your session backend (e.g. delete the `flask_session/` cache dir) so a stale null session isn't reused.
